apps/openmw/mwmechanics/aicast.cpp

```diff
@@ -0,0 +1,84 @@
+#include "aicast.hpp"
+
+#include "../mwbase/environment.hpp"
+#include "../mwbase/mechanicsmanager.hpp"
+
+#include "../mwworld/class.hpp"
+
+#include "aicombataction.hpp"
+#include "creaturestats.hpp"
+#include "spellcasting.hpp"
+#include "steering.hpp"
+
+MWMechanics::AiCast::AiCast(const std::string& targetId, const std::string& spellId, bool isScriptedSpell)
+    : mTargetId(targetId), mSpellId(spellId), mCasting(false), mScripted(isScriptedSpell), mDistance(0)
+{
+    ActionSpell action = ActionSpell(spellId);
+    bool isRanged;
+    mDistance = action.getCombatRange(isRanged);
```

Comment:
Not the point of this PR, but that is a rather bad function. When I first saw this code all my mental alarms went off. Returning a value via return and another value via reference parameter is not good practice. It could return a std::pair, or be split into two functions or if we absolutely must, we could have two parameters (maybe even pointers instead of references so we don't have to add dummy variables for values we don't need).

Reply:
Hm, okay maybe I overreacted a bit. We seem to use return both via return value and parameter quite a bit. Still the dummy variable is bad.